src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.cpp

```diff
@@ -0,0 +1,1850 @@
+                       commandData.maxFrameRate >= 1,
+                   ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
+
+    VerifyOrReturn(commandData.minResolution.width >= 1 && commandData.minResolution.height >= 1,
+                   ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
+
+    VerifyOrReturn(commandData.maxResolution.width >= 1 && commandData.maxResolution.height >= 1,
+                   ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
+
+    VerifyOrReturn(commandData.minBitRate >= 1 && commandData.minBitRate <= commandData.maxBitRate && commandData.maxBitRate >= 1,
+                   ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
+
+    VerifyOrReturn(commandData.minFragmentLen <= commandData.maxFragmentLen &&
+                       commandData.maxFragmentLen <= kMaxFragmentLenMaxValue,
+                   ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
+
+    CameraAVStreamMgmtDelegate::VideoStreamAllocateArgs videoStreamArgs;
+    videoStreamArgs.streamUsage        = commandData.streamUsage;
+    videoStreamArgs.videoCodec         = commandData.videoCodec;
+    videoStreamArgs.minFrameRate       = commandData.minFrameRate;
+    videoStreamArgs.maxFrameRate       = commandData.maxFrameRate;
+    videoStreamArgs.minResolution      = commandData.minResolution;
+    videoStreamArgs.maxResolution      = commandData.maxResolution;
+    videoStreamArgs.minBitRate         = commandData.minBitRate;
+    videoStreamArgs.maxBitRate         = commandData.maxBitRate;
+    videoStreamArgs.minFragmentLen     = commandData.minFragmentLen;
+    videoStreamArgs.maxFragmentLen     = commandData.maxFragmentLen;
+    videoStreamArgs.isWaterMarkEnabled = commandData.watermarkEnabled;
+    videoStreamArgs.isOSDEnabled       = commandData.OSDEnabled;
+
+    // Call the delegate
+    status = mDelegate.VideoStreamAllocate(videoStreamArgs, videoStreamID);
+
+    if (status == Status::Success)
+    {
+        response.videoStreamID = videoStreamID;
+        ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+    }
+    else
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+    }
+}
+
+void CameraAVStreamMgmtServer::HandleVideoStreamModify(HandlerContext & ctx,
+                                                       const Commands::VideoStreamModify::DecodableType & commandData)
+{
+    Status status             = Status::Success;
+    auto & isWaterMarkEnabled = commandData.watermarkEnabled;
+    auto & isOSDEnabled       = commandData.OSDEnabled;
+    auto & videoStreamID      = commandData.videoStreamID;
+
+    // VideoStreamModify should have either the WMARK or OSD feature supported
+    VerifyOrReturn(HasFeature(Feature::kWatermark) || HasFeature(Feature::kOnScreenDisplay),
+                   ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand));
```

Comment:
Why is this checked here and not in the dispatch where Video feature is checked?